cmd/terramate/e2etests/list_test.go

```diff
@@ -195,6 +194,90 @@ terramate {
 	assertRunResult(t, cli.listChangedStacks(), want)
 }
 
+func TestListChangedIgnoreDeletedStackDirectory(t *testing.T) {
+	s := sandbox.New(t)
+
+	stack := s.CreateStack("stack-old")
+	cli := newCLI(t, s.RootDir())
+
+	git := s.Git()
+	git.CommitAll("all")
+	git.Push("main")
+	git.CheckoutNew("deleted-stack")
+
+	test.RemoveAll(t, stack.Path())
+
+	git.CommitAll("removed stack")
+
+	assertRun(t, cli.listChangedStacks())
+}
+
+func TestListChangedIgnoreDeletedNonStackDirectory(t *testing.T) {
+	s := sandbox.New(t)
+
+	s.CreateStack("stack")
+	toBeDeletedDir := filepath.Join(s.RootDir(), "to-be-deleted")
+	test.MkdirAll(t, toBeDeletedDir)
+	test.WriteFile(t, toBeDeletedDir, "test.txt", "")
+	cli := newCLI(t, s.RootDir())
+
+	git := s.Git()
+	git.CommitAll("all")
+	git.Push("main")
+
+	git.CheckoutNew("deleted-diretory")
+
+	test.RemoveAll(t, toBeDeletedDir)
+	git.CommitAll("removed directory")
+
+	assertRun(t, cli.listChangedStacks())
```

Comment:
I dont like this abstraction because `s.RootEntry().CreateDir(path)` is not explicit about where directory is created...

Reply:
In any dir entry on the sandbox we can create files and dirs, why cant the sandbox do it for the root dir ? It seems pretty off to have sandbox manipulating files/dirs inside itself almost all the time..instead of all the time.

Also in general disagree with it not being clear. s.RootEntry returns a DirEntry for the root of the sandbox, creating a dir on it creates a dir on root...pretty clear, just as creating a dir with any DirEntry creates a dir inside it. :thinking:

Reply:
But in the end it is not a huge deal either.